src/components/OrderDropdown/OrderDropdown.tsx

```diff
@@ -0,0 +1,80 @@
+import { ChevronDownIcon } from '@chakra-ui/icons'
+import type { ButtonProps, MenuProps } from '@chakra-ui/react'
+import {
+  Button,
+  Flex,
+  Menu,
+  MenuButton,
+  MenuItemOption,
+  MenuList,
+  MenuOptionGroup,
+  Text,
+} from '@chakra-ui/react'
+import { useCallback, useMemo } from 'react'
+import { useTranslate } from 'react-polyglot'
+
+import { type OrderOption, OrderOptionsKeys } from './types'
+
+type OrderDropdownProps = {
+  value: OrderOptionsKeys
+  onClick: (arg: OrderOptionsKeys) => void
+  buttonProps?: ButtonProps
+} & Omit<MenuProps, 'children'>
+
+const orderOptions: OrderOption[] = [
+  {
+    key: OrderOptionsKeys.ASCENDING,
+    label: 'common.ascending',
+  },
+  {
+    key: OrderOptionsKeys.DESCENDING,
+    label: 'common.descending',
+  },
+]
+
+const width = { base: 'full', md: 'auto' }
+
+const chevronDownIcon = <ChevronDownIcon />
+
+export const OrderDropdown: React.FC<OrderDropdownProps> = ({
+  onClick,
+  value,
+  buttonProps,
+  ...menuProps
+}) => {
+  const translate = useTranslate()
+
+  const renderOptions = useMemo(() => {
+    return orderOptions.map(option => (
```

Comment:
We can just map over `OrderOptionsKeys` values.

Reply:
Done in [1ac600a](https://github.com/shapeshift/web/pull/8006/commits/1ac600aeb95da5729ef78865c837fd0bcec0f777)